dask/_task_spec.py

```diff
@@ -451,6 +455,20 @@ def __init__(self, key: KeyType, target: Alias | TaskRef | KeyType | None = None
     def copy(self):
         return Alias(self.key, self.target)
 
+    def substitute(self, subs):
+        if self.key in subs or self.target.key in subs:
+            sub_key = subs.get(self.key, self.key)
+            val = subs.get(self.target.key, self.target.key)
+            if sub_key == self.key and val == self.target.key:
+                return self
+            if isinstance(val, GraphNode):
+                return val.substitute({}, key=sub_key)
+            return Alias(sub_key, val)
+        return self
```

Comment:
The substitution logic could be factored out into a separate PR on request. It's not used anywhere else so I chose to keep in in here but if it helps with review, I'll break it out